jr-objects/src/main/java/com/fasterxml/jackson/jr/ob/impl/RecordsHelpers.java

```diff
@@ -0,0 +1,85 @@
+package com.fasterxml.jackson.jr.ob.impl;
+
+import com.fasterxml.jackson.jr.ob.impl.POJODefinition.PropBuilder;
+
+import java.lang.reflect.Constructor;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.util.Map;
+
+/**
+ * Helper class to get Java Record metadata.
+ */
+public final class RecordsHelpers {
+    private static boolean supportsRecords;
+
+    private static Method isRecordMethod;
+    private static Method getRecordComponentsMethod;
+    private static Method getTypeMethod;
+
+    static {
+        Method isRecordMethod;
```

Comment:
I think we can avoid this one... see note below.

Reply:
53436e6d0df3d4148186458f03fa4e28436b7350